salt/states/file.py

```diff
@@ -2883,6 +2889,54 @@ def managed(
         the ``source_hash`` parameter.
 
         .. versionadded:: 3005
+
+    signature
+        Ensure a valid GPG signature exists on the selected ``source`` file.
+        Set this to true for inline signatures, or to a file URI retrievable by
+        ``cp.cache_file`` for a detached one.
+
+        .. note::
+
+            This signature will be enforced regardless of source type and will be
+            required on the final output, therefore this does not lend itself well
+            when templates are rendered.
+            The file will not be modified, meaning inline signatures are not
+            removed.
+
+        .. versionadded:: 3007
+
+    source_hash_sig
+        When ``source_hash`` is a file and ``skip_verify`` is not true and ``use_etag``
+        is not true, ensure a valid GPG signature exists on the source hash file.
+        Set this to ``true`` for an inline (clearsigned) signature, or to a file URI
+        retrievable by ``cp.cache_file`` for a detached one. The cached file
```

Comment:
It is the cached `source_hash` file (and its detached signature). Thanks for the comment, I should clear up the wording and highlight that signatures on existing files are not checked - they are assumed to have been verified already.

* `source_hash` file signatures are checked before downloading the managed file's source.
* File signatures are checked after downloading the managed file's source and before its contents are altered, hence a file that is to be created/modified will not be in case the verification fails.